### PR TITLE
Make kmesh cni and manage controller consitent during pod enrollment 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

This is not just a cleanup. It fix some corner cases. Like:

1. retrying add/delete redirection annotation should check pod existence and current status
2. waypoint check should check the "gateway.istio.io/managed" "istio.io-mesh-controller" label instead of `waypoint` name
3. Support "istio.io/dataplane-mode: none" label

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @weli-l to keep an eye on the manage controller
and @lec-bit on the cert manager

File: bpf/include/bpf_helper_defs_ext.h [hunk 1]
```diff
@@ -1,5 +1,18 @@
-/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
-/* Copyright Authors of Kmesh */
+/*
+ * Copyright 2023 The Kmesh Authors.
```
Comment (YaoZengzeng) [inline bpf/include/bpf_helper_defs_ext.h:2]:
```suggestion
 * Copyright 2024 The Kmesh Authors.
```

Comment (hzxuzhonghu) [reply bpf/include/bpf_helper_defs_ext.h:2]: I should revert

File: pkg/utils/enroll.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
+package utils
```
Comment (YaoZengzeng) [inline pkg/utils/enroll.go:1]: copyright

File: bpf/include/bpf_helper_defs_ext.h (hunk 1 above)
Comment (LiZhenCheng9527) [reply bpf/include/bpf_helper_defs_ext.h:2]: suggest remove Year

File: pkg/utils/enroll.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"kmesh.net/kmesh/pkg/constants"
+	"kmesh.net/kmesh/pkg/nets"
+	"kmesh.net/kmesh/pkg/utils/istio"
+)
+
+// ShouldEnroll checks whether a pod should be managed by kmesh.
+// Kmesh manages a pod if a pod has "istio.io/dataplane-mode: kmesh" label
```
Comment (YaoZengzeng) [inline pkg/utils/enroll.go:20]:
```suggestion
// Kmesh manages a pod if a pod has "istio.io/dataplane-mode: Kmesh" label
```

Comment (hzxuzhonghu) [reply pkg/utils/enroll.go:20]: it is case insensitive

**Pull request merged**